### PR TITLE
small changes to min1pipe.m to allow easier use with HPCs

### PR DESCRIPTION
**Pull request by ivosonntag**
small changes to min1pipe.m to circumvent uigetfile in data_info.m
GPU processing is only set if GPU is available (logdemons_unit.m)

### CODE REVIEW

File: min1pipe.m [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function [file_name_to_save, filename_raw, filename_reg] = min1pipe(Fsi, Fsi_new, spatialr, se, ismc, flag)
+function [file_name_to_save, filename_raw, filename_reg] = min1pipe(Fsi, Fsi_new, spatialr, se, ismc, flag, path_name, file_name)
```
Comment (JinghaoLu) [inline min1pipe.m]: better to rename it with min1pipe_hpc

File: demo_min1pipe_HPC.m [hunk 2]
```diff
@@ -0,0 +1,132 @@
+%%% demo of the full MIN1PIPE %%%
+
+addpath(genpath('YOURPATH/MIN1PIPE')) % add MIN1PIPE repo to matlab path
```
Comment (JinghaoLu) [inline demo_min1pipe_HPC.m]: this will prevent the proper installation of cvx toolbox in a fresh machine.

File: min1pipe.m [hunk 3]
```diff
@@ -65,7 +69,7 @@
     %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     
     %% get dataset info %%
-    [path_name, file_base, file_fmt] = data_info;
+    [path_name, file_base, file_fmt] = data_info(file_name, path_name);
```
Comment (JinghaoLu) [inline min1pipe.m]: again, all hpc related functions should be renamed.

File: utilities/elements/data_info.m [hunk 4]
```diff
@@ -1,9 +1,9 @@
-function [path_name, file_base, file_fmt] = data_info
+function [path_name, file_base, file_fmt] = data_info(file_name_tmp, path_name)
```
Comment (JinghaoLu) [inline utilities/elements/data_info.m]: same renaming issue

**Review by JinghaoLu — CHANGES_REQUESTED** [review]
several renaming issue before merging, to avoid conflicts on the regular usage. Thx. Jinghao

**Comment by ivosonntag** [comment]
I reverted the changes you mentioned back and added new functions instead.

**Pull request merged**